[Misc] NFC: Fix random build warnings

davezarzycki opened the pull request:
Unused variables/methods, language extensions, extra semicolons, intentional self assignment, platform specific quirks, etc. Also downright horror shows. For example, I'm not sure that anybody verified that swift-reflection-test.c actually works.

davezarzycki commented:
@swift-ci please smoke test

gottesmm reviewed (commented):
Quick question about the "horror show".

davezarzycki committed:
[Misc] NFC: Fix random build warnings

Unused variables/methods, language extensions, extra semicolons, intentional
self assignment, platform specific quirks, etc.

davezarzycki committed:
[Testing] Fix broken reflection test found via compiler warnings

Note: a usage of PipeMemoryReader_readBytes() was blatantly incorrect.

jrose-apple commented on lib/Frontend/ArgsToFrontendOutputsConverter.cpp:
```diff
@@ -133,7 +133,7 @@ OutputFilesComputer::OutputFilesComputer(
     const FrontendOptions::ActionType requestedAction,
     const llvm::opt::Arg *moduleNameArg, const StringRef suffix,
     const bool hasTextualOutput)
-    : Args(args), Diags(diags), InputsAndOutputs(inputsAndOutputs),
```
Shouldn't the argument be dropped too? cc @davidungar

davezarzycki replied:
Probably. I just wanted to quiet the unused field warning.

davidungar replied:
Yes, please by all means drop the argument, too. (Tx @jrose-apple )

jrose-apple commented on include/swift/SIL/SILInstruction.h:
```diff
@@ -1222,36 +1222,37 @@ struct SILDebugVariable {
+      int_type ArgNo : 16;
     } Data;
-  };
+  } Bits;
```
Why this change?

davezarzycki replied:
Because the following is generated by the default warnings with clang/LLVM top-of-tree:
```
include/swift/SIL/SILInstruction.h:1227:5: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
    struct {
    ^
```

davezarzycki commented:
Hi @gottesmm – Thanks for the feedback. Done.

@swift-ci please smoke test

jrose-apple commented on unittests/runtime/Metadata.cpp:
```diff
@@ -297,12 +297,12 @@ ProtocolDescriptor ProtocolNoWitnessTable{
       auto any = swift_getExistentialTypeMetadata(ProtocolClassConstraint::Any,
                                                   /*superclass=*/nullptr,
-                                                  0, protoList1);
+                                                  0, &proto);
```
Why this change?

davezarzycki replied:
Because the following is generated by the default warnings with clang/LLVM top-of-tree:
```
unittests/runtime/Metadata.cpp:300:44: warning: zero size arrays are an extension [-Wzero-length-array]
  const ProtocolDescriptor *protoList1[] = {};
                                           ^
```

davezarzycki replied:
I briefly looked into that, but I lost interest in verifying that all of the downstream APIs could handle the array itself being null. In theory, none of this matters. Everything should be checking the "num protos" parameter first.

jrose-apple replied:
That's exactly why I'd pass null here. If it doesn't cause any issues, it's a better test, and if it *does* cause issues you found a bug (which you aren't responsible for fixing).

davezarzycki replied:
Well, 10975 tests passed on my Linux box, so let's go with `nullptr`. :-)

davezarzycki commented:
Just to be clear here, I'm only trying to quiet trivial warnings with conservative fixes. I'm not trying to take responsibility for how any particular subsystem should work or interpret any particular dead code as an awkward todo list.

davezarzycki committed:
[Frontend] NFC: Remove unused parameter

davezarzycki commented:
Added @davidungar's feedback.

@swift-ci please smoke test

davezarzycki committed:
[Unit Tests] NFC: Pass nullptr rather than dummy empty array

davezarzycki commented:
@swift-ci please smoke test
